src/main/java/com/gmail/nossr50/skills/repair/RepairManager.java

```diff
@@ -128,6 +133,34 @@ public void handleRepair(ItemStack item) {
 
         // toRemove should be refreshed before the event call.
         toRemove = inventory.getItem(inventory.first(repairMaterial)).clone();
+
+        // Check if we allow enchanted materials to be used to repair objects.
+        // (Servers may provide enchanted items that don't follow their intended use)
+        if (!mcMMO.p.getAdvancedConfig().getAllowEnchantedRepairMaterials()) {
+
+            // See if our proposed item is even enchanted in the first place.
+            if (toRemove.getEnchantments().size() > 0) {
+
+                // Lots of array sorting to gather a possible list of items not including the enchanted item.
+                List<ItemStack> possibleMaterials = Arrays.stream(inventory.getContents())
+                        .filter(Objects::nonNull)
+                        .filter(p -> p.getType() == repairMaterial)
+                        .filter(p -> p.getEnchantments().size() == 0).toList();
```

Comment:
I don't see a reason to make a `List<ItemStack>` here, if the list isn't empty, you only ever access the first element.
Therefore `Stream#findFirst()` makes much more sense here, it returns an `Optional`, allowing you to still check whether a candidate was found or not.

```suggestion
                Optional<ItemStack> possibleMaterial = Arrays.stream(inventory.getContents())
                        .filter(Objects::nonNull)
                        .filter(p -> p.getType() == repairMaterial)
                        .filter(p -> p.getEnchantments().isEmpty())
                        .findFirst();
```

(The `isEmpty` filter can probably be passed as a method reference too but just as a proof of concept)